### PR TITLE
Mask support with CV_Bool in ts and core

### PR DESCRIPTION
**Pull request by asmorkalov**
Partially cover https://github.com/opencv/opencv/issues/25895

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
@opencv-alalek @mshabunin @opencv-alalek Friendly reminder.

File: modules/core/test/test_arithm.cpp [hunk 1]
```diff
@@ -1923,7 +1923,7 @@ TEST(Core_ArithmMask, uninitialized)
 
         Mat a(dims, sizes, type), a1;
         Mat b(dims, sizes, type), b1;
-        Mat mask(dims, sizes, CV_8U);
+        Mat mask(dims, sizes, CV_Bool);
```
Comment (opencv-alalek) [inline modules/core/test/test_arithm.cpp:1926]: We should not modify existed tests. We should add new tests for new types support.

Comment (vpisarev) [reply modules/core/test/test_arithm.cpp:1926]: I would make it random, 8U or bool

File: modules/core/include/opencv2/core.hpp [hunk 2]
```diff
@@ -352,7 +352,7 @@ result of an incorrect sign in the case of overflow.
 @param src2 second input array or a scalar.
 @param dst output array that has the same size and number of channels as the input array(s); the
 depth is defined by dtype or src1/src2.
-@param mask optional operation mask - 8-bit single channel array, that specifies elements of the
+@param mask optional operation mask - CV_8U or CV_Bool single channel array, that specifies elements of the
```
Comment (vpisarev) [inline modules/core/include/opencv2/core.hpp:355]: why not add CV_8S as yet another option, since we are doing changes anyway

Comment (asmorkalov) [reply modules/core/include/opencv2/core.hpp]: Done

File: modules/core/test/test_arithm.cpp (hunk 1 above)
Comment (asmorkalov) [reply modules/core/test/test_arithm.cpp]: Added couple of more tests.

**Pull request merged**